File: TripKitAndroid/src/main/java/com/skedgo/android/tripkit/TripUpdateApi.java
package com.skedgo.android.tripkit;

import skedgo.tripkit.routing.RoutingResponse;

import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.Url;
import rx.Observable;

interface TripUpdateApi {
  @GET Observable<RoutingResponse> fetchUpdateAsync(
      @Url String url,
      @Query("v") String apiVersion
  );
}

File: TripKitAndroid/src/main/java/com/skedgo/android/tripkit/TripUpdateApi.kt
package com.skedgo.android.tripkit

import skedgo.tripkit.routing.RoutingResponse

import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Url
import rx.Observable

internal interface TripUpdateApi {

  @GET
  fun fetchUpdateAsync(
      @Url url: String,
      @Query("includeStops") includeStops: Boolean = true,
      @Query("v") v: Int = 12
  ): Observable<RoutingResponse>
}